lib/hooks.js

```diff
@@ -174,6 +177,88 @@ function hookRunnerApplication (hookName, boot, server, cb) {
   }
 }
 
+function onListenHookRunner (boot, server) {
```

Comment:
I believe that, as the `onListenHookRunner` does not have the intention to stop the server, and is not a `lifecycle` hook or `applicationHook`, it should be ok if we omit the `avvio` linking. At this moment `avvio` already marked the app as `ready` to start listening.

Reply:
So should we just call them through a for loop? I think this gets rid of the need for wrap and next too?

Reply:
Hmm, don't think so.
Basically `next` controls the flow between encapsulated contexts (`fastify` parent instance, and its child through `fastify.register`) and its hooks set, and maybe `wrap` can be refactored to only promisify the function, and/or handle it promise version.